text/0000-buildpack-input-vars.md

```diff
@@ -0,0 +1,142 @@
+# Meta
+[meta]: #meta
+- Name: Replace positional args to Buildpack executables with env vars
+- Start Date: 2021-11-18
+- Author(s): [jkutner](https://github.com/jkutner)
+- Status: Draft Draft
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal to replace the positional arguments of Buildpack executables with named environment variables.
+
+# Definitions
+[definitions]: #definitions
+
+- *positional arguments* - values passed into an executable command such that they are accessible as `$1`, `$2`, etc
+
+# Motivation
+[motivation]: #motivation
+
+Postional arguments to Buildpack executables have been the way for the buildpack execution engine to provide inputs for buildpacks since buildpacks v1 was created. However, positional arguments are limiting, and what they represent is not obvious without reading the spec.
+
+In the spec today they are defined as:
+
+* `/bin/detect <platform[AR]> <plan[E]>`
+* `/bin/build <layers[EIC]> <platform[AR]> <plan[ER]>`
+
+Using env vars helps make these inputs more easily accesible from language bindings like [libcnb.bash](https://github.com/jkutner/libcnb.bash).
+
+# What it is
+[what-it-is]: #what-it-is
+
+We will deprecate the positional arguments to `bin/detect` and `bin/build` with the following environment variables:
+
+### bin/detect
+
+* `CNB_PLATFORM_DIR` - replaces the first positional argument to `bin/build`. Uses the same env var name as the Platform spec.
+* `CNB_PLAN_PATH` - replaces the second positional argument to `bin/build`. Uses the same env var name as the Platform spec.
```

Comment:
This file is a buildpack specific [build plan](https://github.com/buildpacks/spec/blob/main/buildpack.md#build-plan-toml). which is different from the [plan.toml](https://github.com/buildpacks/spec/blob/main/platform.md#plantoml-toml) which is referred to with `CNB_PLAN_PATH` in the platform spec.

This build plan is also different from the [buildpack plan](https://github.com/buildpacks/spec/blob/main/platform.md) we are using `CNB_PLAN_PATH` to reference in the builld interface below.

Using the same name for three different files with three different schemas might be confusing.

Maybe instead we could go with `CNB_BUILD_PLAN_PATH` and `CNB_BP_PLAN_PATH`? Or a better idea if anyone has one?

Reply:
`CNB_BUILD_PLAN_PATH` seems good.

Reply:
done